torch/csrc/inductor/aot_runtime/model.h

```diff
@@ -331,45 +348,40 @@ class AOTInductorModel : public AOTInductorModelBase<AOTInductorModel> {
   }
 };
 
-#define AOTI_TORCH_ERROR_CHECK(call)                                      \
-  if ((call) != AOTI_TORCH_SUCCESS) {                                     \
-    throw std::runtime_error(                                             \
-        std::string(#call " API call failed at ") + __FILE__ + ", line" + \
-        std::to_string(__LINE__));                                        \
-  }
-
-using RAIIAtenTensorHandle = std::shared_ptr<AtenTensorOpaque>;
-
-inline RAIIAtenTensorHandle create_raii_tensor_handle_for_extern(
-    AtenTensorHandle handle) {
+// TODO: will update RAIIAtenTensorHandle to unique_ptr and change this
+inline RAIIAtenTensorHandle create_raii_tensor_handle(AtenTensorHandle handle) {
```

Comment:
It would be great if this had "steal" in its name, because that's what it's doing.

Also, I hate having to hand-review all of the create raii tensor handle call sites. It would be much easier to verify that everything was done correctly if you have a model.so only header that wraps all of the ABI stable aoti functions into C++ functions that do RAII handling. Then, we only need to audit that file, and all other sites are guaranteed to do things correctly by RAII.

Reply:
The wrapper file suggestion is good, but I can do that a later PR.